solidity/contracts/ERC20WithData.sol

```diff
@@ -7,8 +7,21 @@ import '@openzeppelin/contracts/utils/Context.sol';
 import '@openzeppelin/contracts/access/Ownable.sol';
 
 /**
-    @dev Mintable+burnable form of ERC20 with data event support.
-*/
+ * Example ERC20 token with mint, burn, and attached data support.
+ *
+ * This contract demonstrates a very simple ERC20 fungible token. Notes on functionality:
+ *   - the contract owner (ie deployer) is the only party allowed to mint
+ *   - any party can approve another party to manage (ie transfer) a certain amount of their
+ *     tokens (approving for MAX_INT gives an unlimited approval)
+ *   - you may only burn your own tokens
```

Comment:
Maybe `Any party can burn their own tokens` (given I think this is a slightly more unusual option)